rollup/fees/rollup_fee.go

```diff
@@ -216,7 +206,7 @@ func CalculateL1DataFee(tx *types.Transaction, state StateDB, config *params.Cha
 		return big.NewInt(0), nil
 	}
 
-	raw, err := rlpEncode(tx)
+	raw, err := tx.MarshalBinary()
```

Comment:
`MarshalBinary`still encodes the type information, right?

Reply:
Yes. Both `tx.MarshakBinary` and `tx.EncodeRLP` use `tx.encodeTyped` internally, which prepends the type byte. The difference is that the former directly returns `buf.Bytes()`, while the latter returns it RLP-encoded (`rlp.Encode(w, buf.Bytes())`).